### PR TITLE
disable xpack API and monitoring pipeline for serverless

### PR DESCRIPTION
**Pull request by kaisecheng**
## Release notes

Added support for Elasticsearch serverless. 
* Added support for central pipeline management
* Disabled legacy monitoring pipeline as serverless does not support monitoring API

## What does this PR do?

Serverless made some changes to API
1. `/_xpack` is for internal users with service account. Logstash does not have permission to call it. xpack features are affected.
2. `/_monitoring/bulk?` is not available which causes Logstash legacy monitoring unable to send metrics

This commit adds a call to identify serverless cluster before fetching `/_xpack` to verify the license and supported features. 
When it is serverless, the license checker uses hardcoded xpack info instead of calling xpack API. The internal pipeline of monitoring is disabled and logs error when configured to use legacy monitoring.

One concern about the header `Elastic-Api-Version`. Logstash is using LogStash::Outputs::ElasticSearch to communicate with Elasticsearch, which is a dependency to es-output plugin. The http client is wrapped inside the plugin and the caller cannot add header to request or lookup header in response, which limits us to evolve with the project.

## Why is it important/What is the impact to the user?

This change allows Logstash to use central pipeline management in serverless and points user to monitor with metricbeat

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files (and/or docker env variables)
- [x] I have added tests that prove my fix is effective or that my feature works

## Author's Checklist

- [ ]

## How to test this PR locally

Suggest to test against qa serverless endpoint
take the endpoint and api_key from Vault `vault read secret/ci/elastic-logstash/serverless-test`
use `es_host` and `ls_plugin_api_key`
```
xpack.management.enabled: true
xpack.management.elasticsearch.hosts: 
xpack.management.elasticsearch.api_key: 

xpack.monitoring.enabled: true
xpack.monitoring.elasticsearch.hosts: 
xpack.monitoring.elasticsearch.api_key: 
```

## Related issues

- Closes: https://github.com/elastic/ingest-dev/issues/2303
- Closes: https://github.com/elastic/ingest-dev/issues/2284

### CODE REVIEW

**Review by robbavey — COMMENTED** [review]
a few nits

**Review by robbavey — COMMENTED** [review]
Functionality works as described - my only question is whether we should fail Logstash starting at all if monitoring is misconfigured like this? It is a slightly different scenario from a failure to retrieve a license, as this is not something that can change over time, and can only be fixed by either setting up elastic agent, or pointing to a different Elasticsearch.

I've added some stronger text to the suggestion, but I'd be interested on other's feedback

cc @flexitrev, @roaksoax, @jsvd

File: x-pack/lib/monitoring/internal_pipeline_source.rb [hunk 1]
```diff
@@ -73,6 +73,14 @@ def populate_license_state(xpack_info)
             :log_level => :error,
             :log_message => "Monitoring is not available: #{xpack_info.license_type} is not a valid license for this feature."
         }
+      elsif !xpack_info.feature_enabled?(FEATURE) && is_serverless
+        {
+          :state => :error,
+          :log_level => :error,
+          :log_message => "Internal collection for monitoring is enabled in Logstash, but is not supported in the configured version of Elasticsearch.\n"\
+            "Please configure Elastic Agent to monitor Logstash. Documentation can be found at: \n"\
```
Comment (robbavey) [inline x-pack/lib/monitoring/internal_pipeline_source.rb:81]: I'd like to add emphasis that we're not sending monitoring

```suggestion
            "This instance of Logstash will NOT send its monitoring data to Elasticsearch. Please configure Elastic Agent to monitor Logstash. Documentation can be found at: \n"\
```

**Review by robbavey — APPROVED** [review]
LGTM apart from a comment on emphasizing that Logstash will not be monitored.

**Comment by kaisecheng** [comment]
@logstashmachine backport 8.10

**Pull request merged**